scripts/update_datasets.py

```diff
@@ -0,0 +1,170 @@
+import requests
+import json
+import yaml
+
+def read_yaml(file_path):
+    with open(file_path, 'r') as file:
+        return yaml.safe_load(file)
+
+def download_json(url):
+    response = requests.get(url)
+    return json.loads(response.text)
+
+def read_markdown(file_path):
+    with open(file_path, 'r') as file:
+        return file.read()
+
+def write_markdown(file_path, content):
+    with open(file_path, 'a') as file:
+        file.write(content)
+
+def generate_markdown_with_tooltips(json_data, heading_dataset, dataset):
+    markdown_content = "- Variables: \n"
+    
+    if dataset.startswith('cordex'):
+        variable_details = {}
+        for scenario, temporal_resolutions in json_data['data'].items():
+            for temporal_resolution, variables in temporal_resolutions.items():
+                for variable, details in variables.items():
+                    if variable not in variable_details:
+                        variable_details[variable] = {}
+                    if scenario not in variable_details[variable]:
+                        variable_details[variable][scenario] = set()
+                    # Add resolution to the set of resolutions for this scenario
+                    variable_details[variable][scenario].add(temporal_resolution)
+        
+        first_iteration = True
+        for variable, scenarios_resolutions in variable_details.items():
+            markdown_content += process_cordex_data(variable, scenarios_resolutions, first_iteration)
+            first_iteration = False
```

Comment:
Could you split this function into two parts?
1. generate_markdow_for_cordex
2. generate_markdown_for_default

It is hard to find out what the function actually does.

Reply:
I have refactored this function in https://github.com/C2SM/c2sm.github.io/pull/133/commits/3f4e68cab5b200e176f6b76920ef8d035081375c and added docstrings in https://github.com/C2SM/c2sm.github.io/pull/133/commits/5dbf37d3dce81ef4e95e2db93c84822811c0617f. Should be clearer now.